azurelinuxagent/common/event.py

```diff
@@ -339,12 +342,27 @@ def add_metric(self, category, counter, instance, value, log_event=False):
         event.parameters.append(TelemetryEventParam('Instance', instance))
         event.parameters.append(TelemetryEventParam('Value', value))
 
+        self._add_default_parameters_to_event(event, container_id=os.environ.get(
+            CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED"), event_time_opcode_name=event_time_opcode_name,
+                                              event_pid=event_pid, event_tid=event_tid, task_name=task_name,
+                                              keyword_name=keyword_name)
         data = get_properties(event)
         try:
             self.save_event(json.dumps(data))
         except EventError as e:
             logger.error("{0}", e)
 
+    @staticmethod
+    def _add_default_parameters_to_event(event, container_id="UNINITIALIZED", event_time_opcode_name=None, event_pid=0,
+                                         event_tid=0, task_name='', keyword_name=''):
+        event.parameters.append(TelemetryEventParam('ContainerId', container_id))
+        event.parameters.append(TelemetryEventParam('OpcodeName', event_time_opcode_name if event_time_opcode_name
+                                                    else datetime.utcnow().__str__()))
+        event.parameters.append(TelemetryEventParam('EventTid', event_tid))
```

Comment:
shouldn't this be adding the current pid/tid?

Reply:
Do you mean the default values here or the passed values from the top?

For default values, I was thinking -1 for default values as well as PID 0 has a special meaning.